tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -60,12 +61,24 @@ var XTablesCmds = sets.NewSet(
 )
 
 // RealDependencies implementation of interface Dependencies, which is used in production
-type RealDependencies struct{}
+type RealDependencies struct {
+	NetworkNamespace string
+	CNIMode          bool
+}
 
 func (r *RealDependencies) execute(cmd string, redirectStdout bool, args ...string) error {
-	fmt.Printf("%s %s\n", cmd, strings.Join(args, " "))
+	log.Infof("%s %s\n", cmd, strings.Join(args, " "))
+	if r.CNIMode {
+		originalCmd := cmd
+		cmd = constants.NSENTER
+		args = append([]string{fmt.Sprintf("--net=%v", r.NetworkNamespace), "--", originalCmd}, args...)
+	}
 	externalCommand := exec.Command(cmd, args...)
-	externalCommand.Stdout = os.Stdout
+	if r.CNIMode {
+		externalCommand.Stdout = os.Stderr
```

Comment:
Its not clear to me why its different, could use a comment

Reply:
Sorry it is debug left over. I was trying to figure out why CNI network configuration fails during development.